server/seedCards.ts

```diff
@@ -1,28 +1,37 @@
-import { Card, Shape, FillStyle, Color } from './db/card';
+import { Card, Shape, FillStyle, Color } from "./db/card";
 
-const seedCards = async (shapes: Array<Shape>, fills: Array<FillStyle>, colors: Array<Color>, numbers: Array<number>) => {
-    let existingCards = 0;
-    for (const number of numbers) {
-        for (const shape of shapes) {
-            for (const fillStyle of fills) {
-                for (const color of colors) {
-                    try {
-                        const cardExists = await Card.exists({ shape, fillStyle, color, number });
-                        if (cardExists) {
-                            existingCards += 1;
-                            continue;
-                        }
-                        await Card.create({ shape, fillStyle, color, number });
-                    } catch (err) {
-                        if (!(err.code === 11000 && err.name === 'MongoError')) {
-                            throw err;
-                        }
-                    }
-                }
+const seedCards = async (
+  shapes: Array<Shape>,
+  fills: Array<FillStyle>,
+  colors: Array<Color>,
+  numbers: Array<number>
+) => {
+  let existingCards = 0;
+  for (const number of numbers) {
+    for (const shape of shapes) {
+      for (const fillStyle of fills) {
+        for (const color of colors) {
+          try {
+            const cardExists = await Card.exists({
+              shape,
+              fillStyle,
+              color,
+              number,
+            });
+            if (cardExists) {
+              existingCards += 1;
+              continue;
             }
+            await Card.create({ shape, fillStyle, color, number });
```

Comment:
Just because I'm confused as to why there are cards created once on application boot: this is just to populate the possibilities, the cards aren't shared among all games (as in, card A is duplicated and available across games?)

Reply:
The `Card`s are essentially constants in the database that are referenced by games. Each game, when started, will create references to the `Card`s. 12 `Card`s will be referenced in `Room.board`, and the remaining cards will be added into `Room.cardsAvailable`. The alternative was storing really long strings in the database, which I felt was kind of unwieldy and unclean.

Reply:
Sounds good to me!